Sources/ArcGISToolkit/FloorFilter/FloorFilterViewModel.swift

```diff
@@ -0,0 +1,126 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import UIKit
+import ArcGIS
+
+/// View Model class that contains the Data Model of the Floor Filter
+/// Also contains the business logic to filter and change the map extent based on selected site/level/facility
+internal class FloorFilterViewModel {
+    /// The MapView, Map and Floor Manager are set in the FloorFilterViewController when the map is loaded
+    public var mapView: AGSMapView?
+    public var map: AGSMap?
+    public var floorManager: AGSFloorManager?
+    
+    public var sites: [AGSFloorSite] {
+        return floorManager?.sites ?? []
+    }
+    
+    /// Facilities in the selected site
+    /// If no site is selected then the list is empty
+    /// If the sites data does not exist in the map, then use all the facilities in the map
+    public var facilities: [AGSFloorFacility] {
+        guard let floorManager = floorManager else { return [] }
+        return sites.isEmpty ? floorManager.facilities : floorManager.facilities.filter { $0.site == selectedSite }
+    }
+    
+    /// Levels that are visible in the expanded Floor Filter levels table view
+    /// Reverse the order of the levels to make it in ascending order
+    public var visibleLevelsInExpandedList: [AGSFloorLevel] {
+        guard let floorManager = floorManager else { return [] }
+        return facilities.isEmpty ? floorManager.levels : floorManager.levels.filter { $0.facility == selectedFacility }.reversed()
+    }
+    
+    /// All the levels in the map
+    private var allLevels: [AGSFloorLevel] {
+        return floorManager?.levels ?? []
+    }
+    
+    /// The site, facility, and level that are selected by the user
+    public var selectedSite: AGSFloorSite?
+    public var selectedFacility: AGSFloorFacility?
+    public var selectedLevel: AGSFloorLevel?
+    
+    /// The default vertical order is 0 according to Runtime 100.12 update for AGSFloorManager
+    static private let defaultVerticalOrder = 0
+
+    /// Reset FloorFilter saved data
+    public func reset() {
+        floorManager = nil
+        selectedSite = nil
+        selectedFacility = nil
+        selectedLevel = nil
+    }
+    
+    /// Gets the default level for a facility
+    /// Uses level with vertical order 0 otherwise gets the lowest level
+    public func getDefaultLevelForFacility(facility: AGSFloorFacility?) -> AGSFloorLevel? {
+        let candidateLevels = allLevels.filter { $0.facility == facility }
+        return candidateLevels.first { $0.verticalOrder == 0 } ?? getLowestLevel(levels: candidateLevels)
```

Comment:
With the parameter of `getLowestLevel()` removed, this could be simplified to the following:

```swift
allLevels.first { floorLevel in
    floorLevel.facility == facility && floorLevel.verticalOrder == .zero
} ?? getLowestLevel()
```